### PR TITLE
Add hypershift config to the provision shard API and change the management cluster from name to ID

### PR DESCRIPTION
**Pull request by tzvatot**
@samira-barouti @zgalor @pvasant PTAL

### CODE REVIEW

File: model/clusters_mgmt/v1/provision_shard_type.model [hunk 1]
```diff
@@ -38,9 +38,12 @@ class ProvisionShard {
 	// Contains the cloud provider name
 	link CloudProvider CloudProvider
 
-	// Contains the name of the management cluster for Hypershift clusters that are assigned to this shard.
+	// Contains the configuration for Hypershift
+	HypershiftConfig ServerConfig
```
Comment (zgalor) [inline model/clusters_mgmt/v1/provision_shard_type.model:42]: comment should end with `.` 

Also the one above (cloud provider name), which is not part of this PR...

Comment (tzvatot) [reply model/clusters_mgmt/v1/provision_shard_type.model:42]: non of the comments in this file ends with a dot.

Comment (tzvatot) [reply model/clusters_mgmt/v1/provision_shard_type.model:42]: Fixed. PreTAL

**Comment by zgalor** [comment]
LGTM. Why is this WIP?

File: model/clusters_mgmt/v1/provision_shard_type.model (hunk 1 above)
Comment (samira-barouti) [reply model/clusters_mgmt/v1/provision_shard_type.model:42]: Is this the ServerConfig of the Service Cluster? HypershiftConfig is a bit confusing as it suggest that this is the config of the hosted hypershift cluster. Something along the line ServiceClusterConfig might be better.

Comment (tzvatot) [reply model/clusters_mgmt/v1/provision_shard_type.model:42]: Given that CS is cluster service, naming something "service cluster" is very confusing IMO.

From OCM perspective, there are two types of provisioning shards - hive, and hypershift.

That's also the name of the field in CS model and DB.
